client/my-sites/plans/jetpack-plans/product-store/hooks/use-shopping-cart-tracker.ts

```diff
@@ -0,0 +1,63 @@
+import { ResponseCartProduct, useShoppingCart } from '@automattic/shopping-cart';
+import { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import isJetpackCloud from 'calypso/lib/jetpack/is-jetpack-cloud';
+import useCartKey from 'calypso/my-sites/checkout/use-cart-key';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+type EventData = {
+	site_id: number | null;
+	selected_product?: string;
+	number_of_products?: number;
+	[ key: string ]: string | number | null | undefined;
+};
+
+type EventDataOptions = {
+	productSlug?: string;
+	addProducts?: boolean;
+};
+
+// Record tracks are not allowing objects inside track events
+// so this little hack to provide array reduced into object so it can be spread in the event
+const reduceProducts = ( products: ResponseCartProduct[] ) =>
+	products.reduce( ( res, item, index ) => {
+		res[ `cart_product_${ index }` ] = item.product_slug;
+		return res;
+	}, {} as any ); //eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
I see your reasoning here, and I agree that tracking which individual products are in the cart is a useful metric. I wonder, @Automattic/experimentation-platform: does this pattern work well with our analytics tools, or is there another format for recording string arrays that yields easier querying?

Reply:
This probably wouldn't be useful unless we specifically care about the order of products in a cart.

What might be better is something like:
`"cart_has_product_slug_{product_slug}": true`

That way we can select carts with certain product slugs.

It's possible to have both around, but I'd want to make the key very explicit e.g. `cart_index_{index}_has_product_slug`. I'm not sure how useful this would be though.

Reply:
@jessie-ross thanks for this. I assumed it might be that way and current implementation wouldn't help much. As I understood, what Product team want to achieve is to have a separation and visibility on what 'bundles' users create for themselves. So it would look like 15% users purchased `Backup` + `Scan` and 5% users purchased `Videopress` + `Social`. 

However, I'm not totally sure it's achievable, at least with tracks. Maybe you have some suggestions? Thank you.

Reply:
>As I understood, what Product team want to achieve is to have a separation and visibility on what 'bundles' users create for themselves. So it would look like 15% users purchased Backup + Scan and 5% users purchased Videopress + Social.
>
>However, I'm not totally sure it's achievable, at least with tracks. Maybe you have some suggestions? Thank you.

If you want to break down by product-combos in Tracks I think it is possible to hack it to work - you'd need to put everything into a deterministically sorted string, e.g.:
`"product_slugs_concatenated": products.sort().join('-')`

It would be good to still have the individual product_slug booleans however - it would make it easier to do manual analyses.

Reply:
Yes, sounds good! Thank you!